charts/src/main/java/hu/ma/charts/line/LineChart.kt

```diff
@@ -0,0 +1,266 @@
+package hu.ma.charts.line
+
+import android.graphics.Paint
+import androidx.compose.animation.core.animateFloatAsState
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.background
+import androidx.compose.foundation.gestures.detectHorizontalDragGestures
+import androidx.compose.foundation.gestures.detectTapGestures
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.Path
+import androidx.compose.ui.graphics.nativeCanvas
+import androidx.compose.ui.graphics.toArgb
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.LocalDensity
+import hu.ma.charts.line.data.DrawAxis
+import hu.ma.charts.line.data.LineChartData
+import kotlin.math.abs
+import kotlin.math.max
+
+@Composable
+fun LineChart(
+  modifier: Modifier = Modifier,
+  data: LineChartData,
+  onDrillDown: ((x: Int, series: List<LineChartData.SeriesData>) -> Unit)? = null
+) {
+
+  val maxXValues = data.series.maxOf { it.points.maxOf { point -> point.x } }
+
+  val maxYValue = max(data.series.maxOf { it.points.maxOf { point -> point.value } },
+    if (data.yLabels.isNotEmpty()) data.yLabels.maxOf { it.atValue } else 0f)
+
+  var drillDownPoint by remember { mutableStateOf<Float?>(null) }
+
+  val screenWidth = with(LocalDensity.current) {
+    (LocalConfiguration.current.screenHeightDp * density)
+  }
```

Comment:
IIRC better approach would be to use `BoxWithConstraints` so that this value would be the actual widest point